pageserver/src/tenant/secondary/downloader.rs

```diff
@@ -1339,10 +1385,22 @@ async fn init_timeline_state(
         match LayerName::from_str(file_name) {
             Ok(name) => {
                 let remote_meta = heatmap_metadata.get(&name);
+                let last_meta = last_heatmap_metadata.get(&name);
                 match remote_meta {
                     Some(remote_meta) => {
+                        let last_meta_generation_file_size = last_meta
+                            .map(|m| m.metadata.generation_file_size())
+                            .unwrap_or(remote_meta.metadata.generation_file_size());
                         // TODO: checksums for layers (https://github.com/neondatabase/neon/issues/2784)
-                        if local_meta.len() != remote_meta.metadata.file_size {
+                        if remote_meta.metadata.generation_file_size()
+                            != last_meta_generation_file_size
+                        {
+                            tracing::info!(
```

Comment:
I notice that the existing code doesn't actually remove the layer: is it getting removed later as a consequence of not being `touch_layer`'d, or is that a bug we should fix while updating this code?

Reply:
yeah I think it's a bug, going to fix that.